src/annotator/components/toolbar.js

```diff
@@ -116,13 +116,40 @@ export default function Toolbar({
 
   const [toolSize, setToolSize] = useState(medium);
 
-  const [tool, setTool] = useState('pen');
-
   const [color, setColor] = useState('red');
 
   useEffect(() => {
-    setDoodleOptions({ tool: tool, size: toolSize, color: color });
-  }, [toolSize, tool, color, setDoodleOptions]);
+    setDoodleOptions({ tool: 'pen', size: toolSize, color: color });
+  }, [toolSize, color, setDoodleOptions]);
+
+  const eraserSmall = 10;
+  const eraserMedium = 30;
+  const eraserLarge = 60;
+
+  const [eraserSize, setEraserSize] = useState(eraserMedium);
+
+  useEffect(() => {
+    setDoodleOptions({ tool: 'eraser', size: eraserSize });
+  }, [eraserSize, setDoodleOptions]);
+
+  function setSelected(e, optionalClass = '') {
+    let elems = e.target?.parentElement.querySelectorAll(
+      '.annotator-toolbar-button' + optionalClass
+    );
+    [].forEach.call(elems, function (el) {
+      //@ts-ignore
+      el.classList.remove('selected');
+    });
+    e.target?.classList.toggle('selected');
+  }
+
+  function setPen() {
+    setDoodleOptions({ tool: 'pen', size: toolSize, color: color });
+  }
+
+  function setEraser() {
+    setDoodleOptions({ tool: 'eraser', size: eraserSize, color: color });
+  }
```

Comment:
These should probably be folded into the `useEffect` above. Using the setter functions gained from useState, with useEffect, guarantees that our state remains clean. There should also be a useState for 'tool'. That way you can use a single`useEffect(...[tool, toolSize, eraserSize, color])` This would have to have a conditional to switch between the eraserSize and toolSize depending on the tool selected, but I think that it would be cleaner & less code.